lib/fastly_nsq/consumer.rb

```diff
@@ -99,15 +102,25 @@ def empty?
   attr_reader :tls_options
 
   def connect(queue, **options)
-    Nsq::Consumer.new(
-      {
-        nsqlookupd: FastlyNsq.lookupd_http_addresses,
-        topic: topic,
-        channel: channel,
-        queue: queue,
-        max_attempts: max_attempts,
-        **options
-      }.merge(tls_options)
-    )
+    consumers = FastlyNsq.consumer_nsqds
+    lookupd = FastlyNsq.lookupd_http_addresses
+
+    opts = {
+      topic: topic,
+      channel: channel,
+      queue: queue,
+      max_attempts: max_attempts,
+      **options
+    }.merge(tls_options)
+
+    if !consumers.empty?
```

Comment:
I assume if ! over unless because there is not a else unless ;)

Reply:
`unless else` makes my brain hurt! 🤣